specs/APIReview_TextureStream.md

```diff
@@ -0,0 +1,577 @@
+TextureStream
+===============================================================================================
+
+# Background
+Many native apps use a native engine for real-time communication scenarios, which include video capture, networking and video rendering.  However, often, these apps still use WebView or Electron for UI rendering. The separation between real-time video rendering and UI rendering prevents apps from rendering real-time video inside the web contents. This forces apps to render the real-time video on top of the web contents, which is limiting. Rendering video on top constrains the user experience and it may also cause performance problems. We can ask the native apps to use web renderer for video handling because web standard already provides these features through WebRTC APIs. The end developers, however, prefer to use their existing engine such as capturing and composition, meanwhile using WebRTC API for rendering.
+
+# Description
+The proposed APIs will allow the end developers to stream the captured or composed video frame to the WebView renderer where Javascript is able to insert the frame to the page through W3C standard API of Video, MediaStream element for displaying it.
+The API will use the shared GPU texture buffer so that it can minimize the overall cost with regards to frame copy.
+
+The proposed APIs have dependency on the DirectX and its internal attributes such as
+adapter LUID so it supports only Win32/C++ APIs at this time.
+
+# Examples
+
+## Javascript
+
+This is Javascript code common to both of the following samples:

```

Comment:
Please add two subheadings and make full copies of the code for each

### Render video stream in video HTML element

In this sample, the native code generates video in a texture stream, it is sent to JavaScript, which renders it into a video HTML element.

... js
... c++

### Edit video in JavaScript and send back to native

In this sample, the native code generates video in a texture stream, it is sent to JavaScript, JavaScript edits the video, and sends it back to native code.

... js
... c++